Source/WebCore/rendering/RenderBox.cpp

```diff
@@ -5491,9 +5493,19 @@ LayoutBoxExtent RenderBox::scrollPaddingForViewportRect(const LayoutRect& viewpo
         minimumValueForLength(padding.bottom(), viewportRect.height()), minimumValueForLength(padding.left(), viewportRect.width()));
 }
 
-LayoutUnit synthesizedBaselineFromBorderBox(const RenderBox& box, LineDirectionMode direction)
+LayoutUnit synthesizedBaseline(const RenderBox& box, const RenderStyle* parentStyle, LineDirectionMode direction, BaselineSynthesisEdge edge)
```

Comment:
are we expecting a null_ptr parent style here? (judging by the assert below, no...so shouldn't we just pass in a const ref?)

Reply:
Should we move the null check to the caller instead or is there a better way to get this style information? Here my idea was to fallback to the old behavior in the strange scenario where parent style is null for some reason.

IF we move the null check to the caller we could probably still fall back to the old logic outside of this method and be able to pass in a const ref. Thoughts?

Reply:
Changed